pkg/api/types.go

```diff
@@ -307,6 +311,14 @@ type TestStepConfiguration struct {
 	OpenshiftInstallerCustomTestImageClusterTestConfiguration *OpenshiftInstallerCustomTestImageClusterTestConfiguration `json:"openshift_installer_custom_test_image,omitempty"`
 }
 
+type TestStep struct {
+	Name        string               `json:"name,omitempty"`
+	Image       string               `json:"image,omitempty"`
```

Comment:
As this is a `string` and not a `PipelineImageReference` don't we already allow any pull spec here?

Reply:
There are two points in the execution where this is used:

- When configuration is loaded, to determine dependencies between (ci-operator) steps.
- When the pods are created.

The biggest problem is #1. We know at that point the images that are part of the pipeline, but not those that come from `tag_specification`.

For #2, we can always use `pipeline:src`, `stable:tests`, or even a full spec, which is what I've been using, but since #1 requires us to have that information, that should not be required.

Reply:
So we generally operate on the concept that if an image is named `foo` and there exists a `foo` in the images brought in by the `tag_specification`, the built image overrides that which is pulled in. So, we can accept a simple string here and:

 - if it matches something that is built by the pipeline or the `images`, create that link in the steps
 - if it does not, expect it to be provided by the import from `tag-specification`

Reply:
That's what I wanted to hear/read.